rfc/0000-batch-job-groups.rst

```diff
@@ -0,0 +1,1163 @@
+billing tracking is done at the level of an attempt as different
+attempts for the same job can have different resource pricing if the
+VM configurations are different (4 core worker vs 16 core worker).
+
+Billing is tracked by **resources**. A resource is a product (example:
+preemptible n1-standard-16 VM in us-central1) combined with a version
+tag. Each resource has a rate that is used to compute cost when
+multiplied by the usage of the resource. Resource rates are in units
+that are dependent on the type of resource. For example, VM rates are
+measured in mCPU*hours. Each attempt has a set of resources associated
+with it along with their usage in a resource-dependent set of
+units. For example, a 1 core job has a usage value of 1000 (this value
+is in mCPU). To compute the aggregate cost of a job, we sum up all of
+the usages multiplied by the rates and then multiplied by the duration
+the attempt has been running.
+
+=============
+State Diagram
+=============
+
+A job can be in one of the following states:
+
+- Pending: 1+ parent jobs have not completed yet
+- Ready: No pending parent jobs.
+- Creating: Creating a VM for job private jobs.
+- Running: Job is running on a worker VM.
+- Success: Job completed successfully.
+- Failed: Job failed.
+- Cancelled: Job was cancelled either by the system, by the user, or
+  because at least one of its parents failed.
+- Error: Job failed due to an error in creating the container, an out
+  of memory error, or a Batch bug (ex: user tries to use a nonexistent
+  image).
+
+The allowed state transitions are: Pending -> Ready Ready ->
+{Creating, Running, Cancelled} Creating -> {Running, Cancelled}
+Running -> {Success, Failed, Error, Cancelled}
+
+A job's initial state depends on the states of its parent jobs. If it
+has no parent jobs, its initial state is Ready.
+
+A batch can be in one of the following states:
+
+- completed: All jobs are in a completed state {Success, Failed,
+  Error, Cancelled}
+- running: At least one job is in a non-completed state {Pending,
+  Ready, Running}
+
+The batch and job states are critical for database performance and
+must be indexed appropriately.
+
+
+===============
+Batch Front End
+===============
+
+The Batch Front End service (batch) is a stateless web service that
+handles requests from the user. The front end exposes a REST API
+interface for handling user requests such as creating a batch,
+updating a batch, creating jobs in a batch, getting the status of a
+batch, getting the status of a job, listing all the batches in a
+billing project, and listing all of the jobs in a batch. There are
+usually 3 copies of the batch front end service running at a given
+time to be able to handle requests to create jobs in a batch with a
+high degree of parallelism. This is necessary for batches with more
+than a million jobs.
+
+
+Flow for Creating and Updating Batches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I don't think `~` is valid RST which prevents GitHub from rendering this for easy reading. Can you switch to `-`?